modules/xfem/src/base/XFEMGeometricCut2D.C

```diff
@@ -33,12 +33,17 @@ XFEMGeometricCut2D::cutElementByGeometry(const Elem* elem, std::vector<CutEdge>
 
   Real fraction = cutFraction(time);
 
-  unsigned int nsides = elem->n_nodes();
+  unsigned int n_sides = elem->n_sides();
 
-  for (unsigned int i = 0; i < nsides; ++i)
+  for (unsigned int i = 0; i < n_sides; ++i)
   {
-    const Node *node1 = elem->get_node(i);
-    const Node *node2 = elem->get_node(i<(nsides-1) ? i+1 : 0);
+    // This returns the lowest-order type of side, which should always
+    // be an EDGE2 here because this class is for 2D only.
+    UniquePtr<Elem> curr_side = elem->side(i);
+    if (curr_side->type() != EDGE2)
```

Comment:
It actually should work because the side() method returns the lowest-order element that can represent the side, so it should always be an EDGE2. I haven't actually tested it yet.